modules/ganache/Dockerfile

```
@@ -1,6 +1,22 @@
-FROM trufflesuite/ganache:latest
-COPY ganache_with_accounts.sh /app
+#FROM trufflesuite/ganache:latest
+#COPY ganache_with_accounts.sh /app
+#
+#EXPOSE 8545
+#
+#ENTRYPOINT ["/bin/sh", "/app/ganache_with_accounts.sh"]
+
+
+
+
+FROM node:alpine
+
+WORKDIR /var/hardhat_node
+
+COPY hardhat.config.js /var/hardhat_node
+
+RUN cd /var/hardhat_node && npm init -y
+RUN cd /var/hardhat_node && npm i hardhat
 
 EXPOSE 8545
 
-ENTRYPOINT ["/bin/sh", "/app/ganache_with_accounts.sh"]
+CMD ["npx", "hardhat", "node"]
```

The way how the semantics of the Docker entrypoint and command parameters works, is, that the entrypoint defines the command to call upon starting a container, and the commands add the parameters for said command.

This line could also written as:

```suggestion
ENTRYPOINT ["npx"]
CMD ["hardhat", "node"]
```